src/services/services.ts

```diff
@@ -3839,7 +3837,9 @@ module ts {
                 if (symbolKind) {
                     pushTypePart(symbolKind);
                     displayParts.push(spacePart());
-                    addFullSymbolName(symbol);
+                    if (!symbol.isAnonymous) {
+                        addFullSymbolName(symbol);
+                    }

```

Comment:
and if understand correctly, what we are trying to handle here is export default function and class declarations with no name, and this change is looking much bigger than the scope of the original problem. in my mind these two cases can be easily special cased in getQuickInfo.